tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,544 @@
+        /// </summary>
+        public async Task DisposeAsync()
+        {
+            // Don't attempt cleanup if we don't have a store to delete
+            if (_appConfigStore == null && !_shouldDeleteResourceGroup)
+            {
+                return;
+            }
+
+            try
+            {
+                if (_appConfigStore != null)
+                {
+                    await _appConfigStore.DeleteAsync(WaitUntil.Completed);
+                }
+
+                if (_shouldDeleteResourceGroup && _resourceGroup != null)
+                {
+                    await _resourceGroup.DeleteAsync(WaitUntil.Completed);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine($"Test cleanup failed: {ex}. You may need to manually delete the resources: Store={_testStoreName}, ResourceGroup={(_shouldDeleteResourceGroup ? _testResourceGroupName : "N/A")}");
+            }
+        }
+
+        /// <summary>
+        /// Creates a unique prefix for test keys to ensure test isolation
+        /// </summary>
+        private string GetUniqueKeyPrefix(string testName)
+        {
+            // Use a combination of the test prefix and test method name to ensure uniqueness
+            return $"{TestKeyPrefix}_{testName}_{Guid.NewGuid().ToString("N").Substring(0, 8)}";
+        }
+
+        /// <summary>
+        /// Setup test-specific keys and settings
+        /// </summary>
+        private async Task<(string keyPrefix, string sentinelKey, string featureFlagKey)> SetupTestKeys(string testName)
+        {
+            string keyPrefix = GetUniqueKeyPrefix(testName);
+            string sentinelKey = $"{keyPrefix}:Sentinel";
+            string featureFlagKey = $".appconfig.featureflag/{keyPrefix}Feature";
+
+            // Create test-specific settings
+            var testSettings = new List<ConfigurationSetting>
+            {
+                new ConfigurationSetting($"{keyPrefix}:Setting1", "InitialValue1"),
+                new ConfigurationSetting($"{keyPrefix}:Setting2", "InitialValue2"),
+                new ConfigurationSetting(sentinelKey, "Initial"),
+                ConfigurationModelFactory.ConfigurationSetting(
+                    featureFlagKey,
+                    @"{""id"":""" + keyPrefix + @"Feature"",""description"":""Test feature"",""enabled"":false}",
+                    contentType: FeatureManagementConstants.ContentType)
+            };
+
+            // Add test-specific settings to the store
+            foreach (var setting in testSettings)
+            {
+                await _configClient.SetConfigurationSettingAsync(setting);
+            }
+
+            return (keyPrefix, sentinelKey, featureFlagKey);
+        }
+
+        [Fact]
+        public void LoadConfiguration_RetrievesValuesFromAppConfiguration()
+        {
+            Skip.If(_skipTests, _skipReason);
```

Comment:
I don't really understand why we'd want skip. If an exception occurs during initialization, just let it bubble and blow up.

Reply:
Yea that's fair, I was thinking about whether the authentication would still work when running in a github workflow and how to handle that. This still isn't what I originally intended though, so I'll look into it.